engine/env/spark/src/main/java/org/datacleaner/spark/utils/ResultFilePathUtils.java

```
@@ -0,0 +1,86 @@
+    private static final String RESULT_FILE_EXTENSION = FileFilters.ANALYSIS_RESULT_SER.getExtension();
+    
+    /**
+     * Gets the hdfs path of job's result
+     * 
+     * @return
+     */
+    public static String getResultFilePath(final JavaSparkContext sparkContext, final SparkJobContext sparkJobContext) {
+        String resultPath = sparkJobContext.getResultPath();
+        final Configuration hadoopConfiguration = sparkContext.hadoopConfiguration();
+        final String fileSystemPrefix = hadoopConfiguration.get("fs.defaultFS");
+        if (resultPath == null || resultPath.isEmpty()) {
+            resultPath = createPath(fileSystemPrefix, DEFAULT_RESULT_PATH);
+        } else {
+            final URI uri = URI.create(resultPath);
+            if (!uri.isAbsolute()) {
+                resultPath = createPath(fileSystemPrefix, resultPath);
+            }
+        }
+
+        if (!resultPath.endsWith(RESULT_FILE_EXTENSION)) {
+            final String analysisJobXmlName = sparkJobContext.getAnalysisJobName();
+            final Date date = new Date();
+            final String filename = analysisJobXmlName + "-" + date.getTime() + RESULT_FILE_EXTENSION;
+            final String filePath = createPath(resultPath, filename);
+            return filePath;
+        }
+        return resultPath;
+    }
+
+    public static String createPath(final String fileSystemPrefix, String resultPath) {
+
+        final URI uri;
+        try {
+            URI systemPrefix = new URI(fileSystemPrefix);
+            if (!resultPath.startsWith("/")) {
+                resultPath = "/" + resultPath;
+            }
+            uri = new URI("hdfs", null, systemPrefix.getHost(), systemPrefix.getPort(), systemPrefix.getPath()
+                    + resultPath, null, null);
+            return uri.toString();
+        } catch (URISyntaxException e) {
+            logger.error("Error while trying to create url for saving the job", e);
```

Is there any sense in just logging the error here, and continuing with a null value, effectively swallowing the exception? Normally failing fast is better because then the user knows he needs to change something.